mypy/test/testcheck.py

```diff
@@ -278,22 +278,17 @@ def parse_module(self, program_text: str, incremental: int = 0) -> List[Tuple[st
 
     def parse_options(self, program_text: str, testcase: DataDrivenTestCase) -> Options:
         options = Options()
-        flags = re.search('# options: (.*)$', program_text, flags=re.MULTILINE)
-        version_flag = re.search('# pyversion: (.*)$', program_text, flags=re.MULTILINE)
-        platform_flag = re.search('# platform: (.*)$', program_text, flags=re.MULTILINE)
+        flags = re.search('# flags: (.*)$', program_text, flags=re.MULTILINE)
 
+        flag_list = None
         if flags:
-            options_to_enable = flags.group(1).split()
-            for opt in options_to_enable:
-                setattr(options, opt, True)
-
-        # Allow custom pyversion comment to override testcase_pyversion
-        if version_flag:
-            options.python_version = parse_version(version_flag.group(1))
+            flag_list = flags.group(1).split()
+            _, options = process_options(flag_list, require_targets=False)
         else:
-            options.python_version = testcase_pyversion(testcase.file, testcase.name)
+            options = Options()
 
-        if platform_flag:
-            options.platform = platform_flag.group(1)
+        # Allow custom python version to override testcase_pyversion
+        if not flag_list or '--python-version' not in flag_list:
```

Comment:
We should also probably check to see if `-2` and `--py2` are in `flag_list`, since those two flags will also set `options.python_version`.